zbus/src/connection/mod.rs

```diff
@@ -996,41 +996,25 @@ impl Connection {
                                     // destination doesn't matter if no name has been registered
                                     // (probably means name it's registered through external means).
                                     if !names.is_empty() && !names.contains_key(dest) {
-                                        trace!("Got a method call for a different destination: {}", dest);
+                                        trace!(
+                                            "Got a method call for a different destination: {}",
+                                            dest
+                                        );
 
                                         continue;
                                     }
                                 }
                             }
-                            let member = match hdr.member() {
-                                Some(member) => member,
-                                None => {
-                                    warn!("Got a method call with no `MEMBER` field: {}", msg);
-
-                                    continue;
-                                }
-                            };
```

Comment:
Seems this is now checked only **after** the spawn? This could mean that a misbehaving client can get us to spawn lots of tasks needlessly. Let's avoid that and check the message integrity etc before spawning the task?

Reply:
* Member name is mandatory for method call messages so "unknown member" means an invalid message here.
* Regardless of this distinction, the point is that we should ensure we've all essentials in the method call before spawning a task for its dispatch.

Reply:
Because the way Interface::call{_mut} is designed, the return value needs to hold the lock. (in contrast with my proposal from https://github.com/dbus2/zbus/issues/674).

So if we want to avoid multiple redundant lookups, we need to do the Method resolution in the spawned task, we can't easily gather all the details before.

Reply:
(Msg and Header can become 'static,  but not 'a in Header<'a>)

Reply:
> But it fails the borrow-checker that it isn't able to match the lifetime of Msg and Header<'_> to 'static when moving those to a future.

Right, I guess because of `spawn` requirements. Still, as I said, I'd rather pay the small[1] penalty of header getter.

---

1. [As documented](https://docs.rs/zbus/latest/zbus/message/struct.Message.html#method.header), this method is no longer costly, just allocates a vec for the fields.

Reply:
> I am not sure what you want to do.

I don't really see a problem with what you've already in your latest commit. So I'm as confused as you about what we're discussing here anymore. :)

Reply:
Can you do it on top? I feel stupid, I don't get what you want.